bookkeeper-server/src/main/java/org/apache/bookkeeper/client/RackawareEnsemblePlacementPolicyImpl.java

```diff
@@ -531,26 +561,40 @@ public void handleBookiesThatJoined(Set<BookieSocketAddress> joinedBookies) {
         return networkLocs;
     }
 
+    /*
+     * this method should be called in readlock scope of 'rwLock'
+     */
+    protected Set<BookieSocketAddress> addDefaultZoneBookiesIfMinNumRacksIsEnforced(
+            Set<BookieSocketAddress> excludeBookies) {
+        Set<BookieSocketAddress> comprehensiveExclusionBookiesSet;
+        if (enforceMinNumRacksPerWriteQuorum) {
+            comprehensiveExclusionBookiesSet = new HashSet<BookieSocketAddress>(excludeBookies);
+            topology.getLeaves(getDefaultRack()).forEach((Node node) -> {
+                if (node instanceof BookieNode) {
+                    BookieNode bookieNode = (BookieNode) node;
+                    comprehensiveExclusionBookiesSet.add(bookieNode.getAddr());
```

Comment:
I would like to have INFO message here with the list of all bookies that are in the default rack and are being excluded. I understand it could amplify logging but I am OK to take it when the configuration is asking for strict placement.

Reply:
will do